tests/test_challenges.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for the reauth module."""
+
+import base64
+import json
+import os
+import unittest
+
+import mock
+
+from google_reauth import challenges
+
+import pyu2f
+
+
+class ChallangesTest(unittest.TestCase):
+    """This class contains tests for reauth challanges. """
+
+    # This U2F mock is made by looking into the implementation of the class and
+    # making the minimum requirement to actually answer a challenge.
+    class _U2FInterfaceMock(object):
+        def Authenticate(self, unused_app_id, challenge, unused_registered_keys):
+            raise self.error
+
+    def StartPatch(self, *args, **kwargs):
```

Comment:
use `patch` as a context manager or decorator instead of doing this

Context manager:

```
with mock.patch(...) as thing:
    pass
```

decorator:

```
@mock.patch(thing)
def testMethod(self, thing)
```